### PR TITLE
wireshark4: update to 4.2.0

### PR DESCRIPTION
**Pull request by i0ntempest**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by ghosthound** [comment]
4.2.0 is a major version release, in the past we've made a new port for a new major version of wireshark - not every time, but we do have a few different wireshark* ports.
I lean towards taking this change and having wireshark4 track the latest wireshark-4.x but would like to leave this open for at least a few days to gather feedback from others.

**Comment by pmetzger** [comment]
Having it track seems fine to me.

File: net/wireshark4/Portfile [hunk 1]
```diff
@@ -88,7 +89,7 @@ variant qt5 conflicts no_gui description {Build wireshark with a qt5 GUI} {
 
     configure.args-replace  -DENABLE_APPLICATION_BUNDLE=OFF -DENABLE_APPLICATION_BUNDLE=ON
     configure.args-replace  -DBUILD_wireshark=OFF -DBUILD_wireshark=ON
-    configure.args-append   -DENABLE_QT5=ON
+    configure.args-append   -DUSE_qt6=OFF
```
[the remark this answers is not in the document]
Comment (pmetzger) [reply net/wireshark4/Portfile:92]: I'm confused. It removes the line that says ENABLE_QT5=ON and adds one saying that USE_qt6=OFF, i.e. going from QT5 being ON to no such line being present and a line saying qt6 is OFF.

**Comment by pmetzger** [comment]
@ghosthound Can we commit this now?

**Pull request merged**